src/CLI.hs

```diff
@@ -183,42 +185,43 @@ coq' f = do
     TIO.putStr $ coq claims
 
 
-hevm :: FilePath -> Text -> Maybe FilePath -> Maybe ByteString -> Solvers.Solver -> Maybe Integer -> Bool -> IO ()
-hevm actspec cid sol' code' solver' timeout _ = do
+hevm :: FilePath -> Text -> Maybe FilePath -> Maybe ByteString -> Maybe ByteString -> Solvers.Solver -> Maybe Integer -> Bool -> IO ()
+hevm actspec cid sol' code' initcode' solver' timeout debug' = do
   specContents <- readFile actspec
-  bytecode <- getBytecode
+  (initcode'', bytecode) <- getBytecode
   let act = validation (\_ -> error "Too bad") id (enrich <$> compile specContents)
-  let actbehvs = translateAct act
-  sequence_ $ flip fmap actbehvs $ \(name,behvs,calldata) ->
-    Solvers.withSolvers solver' 1 (naturalFromInteger <$> timeout) $ \solvers -> do
-      solbehvs <- removeFails <$> getBranches solvers bytecode calldata
-      putStrLn $ "\x1b[1mChecking behavior \x1b[4m" <> name <> "\x1b[m of Act\x1b[m"
-      -- equivalence check
-      putStrLn "\x1b[1mChecking if behaviour is matched by EVM\x1b[m"
-      checkResult =<< checkEquiv solvers debugVeriOpts solbehvs behvs
-      -- input space exhaustiveness check
-      putStrLn "\x1b[1mChecking if the input spaces are the same\x1b[m"
-      checkResult =<< checkInputSpaces solvers debugVeriOpts solbehvs behvs
-
-  -- ABI exhaustiveness sheck
+  let opts = if debug' then debugVeriOpts else defaultVeriOpts
+
   Solvers.withSolvers solver' 1 (naturalFromInteger <$> timeout) $ \solvers -> do
-    putStrLn "\x1b[1mChecking if the ABI of the contract matches the specification\x1b[m"
-    checkResult =<< checkAbi solvers debugVeriOpts act bytecode
+    -- Constructor check
+    checkConstructors solvers opts initcode'' bytecode act
+    -- Behavours check
+    checkBehaviours solvers opts bytecode act
+    -- ABI exhaustiveness sheck
+    checkAbi solvers opts act bytecode
 
   where
-    removeFails branches = filter isSuccess $ branches
-
-    isSuccess (EVM.Success _ _ _ _) = True
-    isSuccess _ = False
-
+    getBytecode :: IO (BS.ByteString, BS.ByteString)
     getBytecode =
-      case (sol', code') of
-        (Just f, Nothing) -> do
+      case (sol', code', initcode') of
+        (Just f, Nothing, Nothing) -> do
           solContents  <- TIO.readFile f
-          fmap fromJust $ solcRuntime cid solContents
-        (Nothing, Just c) -> pure $ Format.hexByteString "" c
-        (Nothing, Nothing) -> error "No EVM input is given. Please provide a Solidity file or EVM bytecode"
-        (Just _, Just _) -> error "Both Solidity file and bytecode are given. Please specify only one."
+          bytecodes cid solContents
+        (Nothing, Just c, Just i) -> pure (i, c)
+        (Nothing, Nothing, _) -> error "No runtime code is given"
```

Comment:
might be a bit nicer to print an error message and then exit cleanly with `exitFailure 1` here. It's not such a nice ux to dump a stack trace if the user forgets to pass some needed parameter

Reply:
Done! Also removed an other similar error in the file.